static/js/filter.js

```diff
@@ -99,6 +99,14 @@ function message_matches_search_term(message, operator, operand) {
     case 'sender':
         return people.id_matches_email_operand(message.sender_id, operand);
 
+    case 'huddle-with':
+        var operand_id = people.get_user_id(operand);
+        if (!operand_id) {
+            return false;
+        }
+
+        return (operand_id === message.sender_id);
```

Comment:
I am afraid that is not what these line of code do.
` var operand_id = people.get_user_id(operand);`
This gets the user_id of bob, let it be 5.

`operand_id === message.sender_id`
This checks if this huddle message was sent by user_id:5, that is bob.